src/main/java/securbank/controller/EmployeeController.java

```diff
@@ -179,4 +193,147 @@ public String deleteRequest(@PathVariable UUID requestId, @ModelAttribute Modifi
 		
         return "redirect:/employee/user/request";
     }	
+	
+	@GetMapping("/employee/transactions")
+    public String getTransactions(Model model) {
+
+		List<Transaction> transactions = transactionService.getTransactionsByStatus("Pending");
+		if (transactions == null) {
+			return "redirect:/error?code=500";
+		}
+		model.addAttribute("transactions", transactions);
+		logger.info("GET request:  All pending transactions");
+		
+        return "employee/pendingtransactions";
+    }
+	
+	@GetMapping("/employee/transaction/{id}")
+    public String getTransactionRequest(Model model, @PathVariable() UUID id) {
+		Transaction transaction = transactionService.getTransactionById(id);
+		
+		if (transaction == null) {
+			return "redirect:/error?code=404&path=request-invalid";
+		}
+
+		// checks if employee is authorized for the request to approve
+		if (!transaction.getAccount().getUser().getType().equals
+				("external")) {
+			logger.warn("GET request: Employee unauthrorised request access");
+			return "redirect:/error?code=401&path=request-unauthorised";
+		}
+		model.addAttribute("transaction", transaction);
+		logger.info("GET request: Employee external transaction request by ID");
+		
+        return "employee/approvetransaction";
+    }
+	
+	@PostMapping("/employee/transaction/request/{id}")
+    public String approveRejectTransactions(@ModelAttribute Transaction trans, @PathVariable() UUID id, BindingResult bindingResult) {
+		
+		Transaction transaction = transactionService.getTransactionById(id);
+		if (transaction == null) {
+			return "redirect:/error?code=404&path=request-invalid";
+		}
+		
+		// checks if employee is authorized for the request to approve
+		if (!transaction.getAccount().getUser().getType().equalsIgnoreCase
+				("external")) {
+			logger.warn("GET request: Employee unauthrorised request access");
+					
+			return "redirect:/error?code=401&path=request-unauthorised";
+		}
+		
+		if("approved".equalsIgnoreCase(trans.getApprovalStatus())){
+			if(transactionService.isTransactionValid(transaction)==false){
```

Comment:
I think this should be `if(transactionService.isTransactionValid(transaction) ==false && transaction.getType().equals("DEBIT"))`